Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -17,152 +19,89 @@ class BTAnalyticsService: Equatable {
 
     // MARK: - Private Properties
     
-    private static let events = BTAnalyticsEventsStorage()
+    private let events = BTAnalyticsEventsStorage()
     
     /// Amount of time, in seconds, between batch API requests sent to FPTI
     private static let timeInterval = 15
     
-    private static let timer = RepeatingTimer(timeInterval: timeInterval)
+    private let timer = RepeatingTimer(timeInterval: timeInterval)
 
-    private let apiClient: BTAPIClient
-    
+    private var apiClient: BTAPIClient?
+            
     // MARK: - Initializer
-
-    init(apiClient: BTAPIClient) {
+    
+    private init() { }
+    
+    /// Used to inject `BTAPIClient` dependency into `BTAnalyticsService` singleton
+    func setAPIClient(_ apiClient: BTAPIClient) {
         self.apiClient = apiClient
         self.http = BTHTTP(authorization: apiClient.authorization, customBaseURL: Self.url)
         
-        Self.timer.eventHandler = { [weak self] in
+        timer.eventHandler = { [weak self] in
             guard let self else { return }
             Task {
                 await self.sendQueuedAnalyticsEvents()
             }
         }
-        Self.timer.resume()
+
+        timer.resume()
     }
 
     // MARK: - Deinit
 
     deinit {
-        Self.timer.suspend()
+        timer.suspend()
     }
 
     // MARK: - Internal Methods
     
     /// Sends analytics event to https://api.paypal.com/v1/tracking/batch/events/ via a background task.
-    /// - Parameters:
-    ///   - eventName: Name of analytic event.
-    ///   - correlationID: Optional. CorrelationID associated with the checkout session.
-    ///   - endpoint: Optional. The endpoint of the API request send during networking requests.
-    ///   - endTime: Optional. The end time of the roundtrip networking request.
-    ///   - errorDescription: Optional. Full error description returned to merchant.
-    ///   - isVaultRequest: Optional. If the Venmo or PayPal request is being vaulted.
-    ///   - linkType: Optional. The type of link the SDK will be handling, currently deeplink or universal.
-    ///   - payPalContextID: Optional. PayPal Context ID associated with the checkout session.
-    ///   - startTime: Optional. The start time of the networking request.
-    func sendAnalyticsEvent(
-        _ eventName: String,
-        connectionStartTime: Int? = nil,
-        correlationID: String? = nil,
-        endpoint: String? = nil,
-        endTime: Int? = nil,
-        errorDescription: String? = nil,
-        isVaultRequest: Bool? = nil,
-        linkType: String? = nil,
-        payPalContextID: String? = nil,
-        requestStartTime: Int? = nil,
-        startTime: Int? = nil
-    ) {
+    /// - Parameter event: A single `FPTIBatchData.Event`
+    func sendAnalyticsEvent(_ event: FPTIBatchData.Event) {
```

Comment:
We have those as 2 separate functions for the purpose of unit testing right now. Since `sendAnalyticsEvent` launches an async background task, there is no way for our tests to know when it completes to run it's assertions. `performEventRequest` is acting as an "internal" version that delivers the async result, which we're calling in our unit tests.

We used this pattern in PPCP too to unit test background-type tasks. Of course open to future suggestions here for improving our tests!